ethcore/machine/src/executive.rs

```diff
@@ -407,6 +421,8 @@ impl<'a> CallCreateExecutive<'a> {
 
 					Self::check_static_flag(&params, self.static_flag, self.is_create)?;
 					state.checkpoint();
+
+					let initial_builtin_balance = state.balance(&params.code_address)?;
```

Comment:
Could this state lookup move into the `else` clause at line 454?

Reply:
I'm thinking in the case of an prestate `addr=0x3 balance=0` and then a reverting call to `'addr=0x3 value=1`, the balance in L425 is `0`  and in L454 is `1` (the `Self::transfer_exec_balance` is done). We need to check the balance before the transfer, not after it.

Reply:
Got it, makes sense.